src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClient.java

```diff
@@ -0,0 +1,17 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketDefaultBranch;
+
+import java.util.stream.Stream;
+
+public interface BitbucketBranchClient {
+
+    /**
+     * Hits the bitbucket 'find branches' REST endpoint.
+     * Retrieve the branches matching the supplied filterText param.
+     * Default filterTest is empty. Which gets all branches.
+     * The authenticated user must have REPO_READ permission for the specified repository to call this resource.
+     * @return
+     */
+    Stream<BitbucketDefaultBranch> getRemoteBranches();
```

Comment:
Any thoughts about adding this as a new method under `BitbucketRepositoryClient` as it's technically just another operation on the repository?

Reply:
We could do this, as all branch operations in the rest client fall under the repository, However we'd need to have a second NextPageFetcherImpl underneath the BitbucketPullRequest, and i can see that getting messy quickly

Reply:
We can probably make the `NextPageFetcherImpl` generic instead of implementing it specifically to `BitbucketPullRequest`. In any case, I don't have any objections to having a separate client either. Just making sure we've explored options.

Reply:
Just a note. Explored making NextPageFetcherImpl generic. Currently not possible without a big rework.  

We are passing a `TypeReference<T> returnType` reference here: https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/blob/master/src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRequestExecutor.java#L89

But from Jackson's point of view `BitbucketPage<T>` -> `<T> extends Object` so it binds it as a map instead of a `BitbucketPullRequest`.